contrib/mesos/pkg/executor/service/service.go

```
@@ -86,188 +76,164 @@ func (s *KubeletExecutorServer) AddFlags(fs *pflag.FlagSet) {
+	staticPodsConfigPath string, apiclient *client.Client) error {
+	kcfg, err := s.KubeletConfig()
+	if err == nil {
+		// apply Messo specific settings
+		executorDone := make(chan struct{})
+		kcfg.Builder = func(kc *app.KubeletConfig) (app.KubeletBootstrap, *kconfig.PodConfig, error) {
+			k, pc, err := app.CreateAndInitKubelet(kc)
+			if err != nil {
+				return k, pc, err
+			}
 
-	diskSpacePolicy := kubelet.DiskSpacePolicy{
-		DockerFreeDiskMB: s.LowDiskSpaceThresholdMB,
-		RootFreeDiskMB:   s.LowDiskSpaceThresholdMB,
-	}
+			klet := k.(*kubelet.Kubelet)
 
-	//TODO(jdef) intentionally NOT initializing a cloud provider here since:
-	//(a) the kubelet doesn't actually use it
-	//(b) we don't need to create N-kubelet connections to zookeeper for no good reason
-	//cloud := cloudprovider.InitCloudProvider(s.CloudProvider, s.CloudConfigFile)
-	//log.Infof("Successfully initialized cloud provider: %q from the config file: %q\n", s.CloudProvider, s.CloudConfigFile)
+			s.kletLock.Lock()
+			s.klet = klet
+			s.kletLock.Unlock()
 
-	hostNetworkSources, err := kubelet.GetValidatedSources(strings.Split(s.HostNetworkSources, ","))
-	if err != nil {
-		return err
-	}
+			// decorate kubelet such that it shuts down when the executor is
+			decorated := &executorKubelet{
+				Kubelet:      klet,
+				kubeletDone:  kubeletDone,
+				executorDone: executorDone,
+			}
 
-	hostPIDSources, err := kubelet.GetValidatedSources(strings.Split(s.HostPIDSources, ","))
-	if err != nil {
-		return err
-	}
+			return decorated, pc, nil
+		}
+		kcfg.DockerDaemonContainer = "" // don't move the docker daemon into a cgroup
+		kcfg.Hostname = kcfg.HostnameOverride
+		kcfg.KubeClient = apiclient
+		kcfg.NodeName = kcfg.HostnameOverride
+		kcfg.PodConfig = kconfig.NewPodConfig(kconfig.PodConfigNotificationIncremental, kcfg.Recorder) // override the default pod source
+		kcfg.StandaloneMode = false
+		kcfg.SystemContainer = "" // don't take control over other system processes.
+
```

I add a panic when it's set to fail hard and early.